src/topics/inputs.md

```diff
@@ -143,13 +143,91 @@ The `baseCommand` field will always appear in the final command line before the
 
 [touch]: http://www.linfo.org/touch.html
 
+### Documenting input parameters using `label` or `doc`
+
+`label` is a short, human-readable description for the parameter.
+
+```cwl
+inputs:
+  example_file:
+    type: File
+    label: Use case for label parameter
+    inputBinding:
+      position: 1
+```
+
+`doc` is a documentation string for the object, or an array of strings which should be concatenated.
+
+```cwl
+inputs:
+  example_file:
+    type: File
+    doc: This section demonstrates Optional fields on the Input Parameter 
+    inputBinding:
+      position: 1
+ ```
+
+ ````{note}
+ `label` and `doc` fields can also be used to document output parameters.
+ ````
+
+### Attaching Secondary files to the inputs file
+
+Secondary files are input files that should be staged in the same directory as the input files.
+The `secondaryFiles` parameter defines files or directories that can be automatically uploaded with the primary input file.
+For example, if the primary input is an HTML file (.html), an image file (.png or .jpg) could be included as a secondary file.
+ 
+The use of secondary files are optional for inputs. However, if secondary files are defined for an input file,
+they should exist. An implementation may fail workflow execution if an expected secondary file does not exist.
+
+When defining primary and secondary files in your CWL workflow, 
+use the input parameter to specify your primary files, and use the `secondaryFiles` parameter to define your secondary files.
+An implementation must include matching files and directories in the `secondaryFiles` parameter of the primary file.
+The following example demonstrates the `secondaryFiles` input parameter.
+
+```cwl
+inputs:
+  example_file:
+    type: File
+    inputBinding:
+      prefix: -e
+    secondaryFiles:
+      - .idx
+```
+
+If a value in `secondaryFiles` is a string that is not an expression, 
+it specifies that the following pattern should be applied to the path of the primary file 
+to yield a filename relative to the primary File:
+
+- If the string begins with one or more caret "^" characters, for each caret, remove the last file extension from the path(the last period `.` and all following characters), and append the remainder of the string to the end of the file path.
+
+```cwl
+inputs:
+  example_file:
+    type: File
+    inputBinding:
+      prefix: -e
+    secondaryFiles: 
+      - ^.idx
+```
+
+assuming the primary file above has a .txt extension (example_file.txt), the secondary file will be "example_file.idx".
```

Comment:
Filenames should be be surrounded by `. So `example_file.txt` and `example_file.idx`